core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -221,4 +227,61 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
         }
         return builder;
     }
+
+    @Override
+    public TermSuggestionBuilder doReadFrom(StreamInput in) throws IOException {
+        TermSuggestionBuilder builder = new TermSuggestionBuilder();
+        return builder.suggestMode(in.readOptionalString())
+                      .setAccuracy(in.readOptionalFloat())
+                      .sort(in.readOptionalString())
+                      .stringDistance(in.readOptionalString())
+                      .maxEdits(in.readOptionalVInt())
+                      .maxInspections(in.readOptionalVInt())
+                      .maxTermFreq(in.readOptionalFloat())
+                      .prefixLength(in.readOptionalVInt())
+                      .minWordLength(in.readOptionalVInt())
+                      .minDocFreq(in.readOptionalFloat());
+    }
+
+    @Override
+    public void doWriteTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(suggestMode);
+        out.writeOptionalFloat(accuracy);
+        out.writeOptionalString(sort);
+        out.writeOptionalString(stringDistance);
+        out.writeOptionalVInt(maxEdits);
+        out.writeOptionalVInt(maxInspections);
+        out.writeOptionalFloat(maxTermFreq);
+        out.writeOptionalVInt(prefixLength);
+        out.writeOptionalVInt(minWordLength);
+        out.writeOptionalFloat(minDocFreq);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (other == null || other.getClass() != this.getClass() || !super.equals(other)) {
+            return false;
+        }
+        TermSuggestionBuilder o = (TermSuggestionBuilder) other;
```

Comment:
We would need a call to super() somewhere here. Or, like we did with the query builders, have the superclass declare two abstract doEquals/doHashCode methods that it then calls and the subclasses need to implement.